### PR TITLE
[NPU][A5] Supoort MXFP4-W4A8 for Ascend NPU

### PR DESCRIPTION
**Pull request by LinyuanLi0046**
## Motivation

Add Ascend NPU support for ModelSlim W4A8_MXFP4 path of sglang .

## Modifications

- Extend ModelSlimConfig to recognize W4A8_MXFP4 for both linear and MoE layers in the existing modelslim scheme dispatch path.
- Add new ModelSlim W4A4_MXFP4 schemes:
-  W4A8_MXFP4 linear layers
-  W4A8_MXFP4 MOE layers

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: **Missing `run-ci` label** -- add it to run CI tests.
Latest PR Test (Extra): :x: **Blocked** -- `run-ci` is required first.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py [hunk 1]
```diff
@@ -180,6 +183,7 @@ def get_linear_scheme(
             ("W4A4_DYNAMIC", ModelSlimW4A4Int4),
             ("W8A8", ModelSlimW8A8Int8),
             ("W8A8_DYNAMIC", ModelSlimW8A8Int8),
+            ("W4A8_MXFP", ModelSlimW4A8MxFp),
```
Comment (LinyuanLi0046) [inline python/sglang/srt/layers/quantization/modelslim/modelslim.py:190]: The PR description says this path supports W4A8_MXFP4, but the actual scheme registration uses W4A8_MXFP

File: python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py [hunk 2]
```diff
@@ -141,3 +146,49 @@ def apply(
             bias=bias,
             output_dtype=original_dtype,
         )
+
+
+class NPUW4A8MxFpLinearMethod(_NPULinearMethodBase):
+
+    def process_weights_after_loading(self, layer: torch.nn.Module):
+        layer.weight.data = torch_npu.npu_format_cast(
```
Comment (LinyuanLi0046) [inline python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py:292]: Several transpose() calls in process_weights_after_loading() are not followed by .contiguous(). Since transpose() commonly returns a non-contiguous view, passing such tensors directly to NPU kernels may cause implicit copies or layout-related failures on some versions.

File: python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py [hunk 3]
```diff
@@ -202,6 +207,189 @@ def npu_fused_experts(
     return final_hidden_states
 
 
+def npu_fused_experts_w4a8_mxfp(
```
Comment (LinyuanLi0046) [inline python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py:810]: The prefill and decode MoE paths contain large duplicated blocks: routing setup, two GMM calls, SwiGLU, and unpermute/finalization are structurally very similar. This violates the project style guidance to avoid large duplicated code blocks and increases future maintenance cost. Please extract shared helpers for routing preparation, one MXFP GMM step, and final merge/unpermute logic. Priority: medium.

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py (hunk 1 above)
Comment (LinyuanLi0046) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py:190]: W4A8_MXFP refers to W4A8_MXFP4 in modelslim quant

File: python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py (hunk 2 above)
Comment (LinyuanLi0046) [reply python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py:292]: can’t .contiguous() due to kernel's restrict

File: python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py (hunk 3 above)
Comment (LinyuanLi0046) [reply python/sglang/srt/hardware_backend/npu/quantization/fused_moe_method_npu.py:810]: can refactor later
